### PR TITLE
Add a stable Helm chart for MetalLB.

### PR DESCRIPTION
**Pull request by danderson**
The chart installs MetalLB 0.4.0, which was just released.

### CODE REVIEW

**Comment by danderson** [comment]
I've tried to follow the various guidelines and requirements for stable charts, and I _think_ I got all of them.

While the chart does `helm install` correctly with no other parameters, the resulting installation doesn't do anything because MetalLB needs a config of its own. The chart's NOTES.txt warns of this, per the guidelines. The chart also has a couple of ways to pass in a configuration, to get a fully functioning Helm-managed install.

One thing I was unsure about is that the chart currently hardcodes the `metallb-system` namespace. My rationale here is that MetalLB is a cluster-wide infrastructure service. You cannot install multiple instances simultaneously, that's an entirely unsupported configuration. Given that, it doesn't make sense to me to allow configuring the namespace - and in fact the `metallb-system` namespace is currently hardcoded in the binaries in a few places. WDYT? I'm unfamiliar with Helm best practices when it comes to "cluster singleton" systems such as this.

File: stable/metallb/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+version: 0.1.0
+
+name: metallb
+appVersion: 0.4.1
+description: MetalLB is a load-balancer implementation for bare metal Kubernetes clusters
+keywords: ["load-balancer", "balancer", "lb", "bgp", "arp", "vrrp", "vip"]
+home: https://metallb.universe.tf
+icon: https://metallb.universe.tf/images/logo.png
+maintainers:
+- name: David Anderson
```
Comment (unguiculus) [inline stable/metallb/Chart.yaml]: Use Github username

File: stable/metallb/templates/config.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{{- if (or .Values.config .Values.arpCIDR) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: metallb-system
+  name: config
```
Comment (unguiculus) [inline stable/metallb/templates/config.yaml]: Add standard labels and use the `fullname` template for names. Apply everywhere.

File: stable/metallb/templates/namespace.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: metallb-system
```
Comment (unguiculus) [inline stable/metallb/templates/namespace.yaml]: Why do you hard-code the namespace and create it explicitly. Is this a requirement for MetalLB?

File: stable/metallb/templates/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs: ["get", "list", "watch"]
+- apiGroups: [""]
+  resources: ["events"]
+  verbs: ["create"]
+---
+
+## Service accounts
```
Comment (unguiculus) [inline stable/metallb/templates/rbac.yaml]: Please move service accounts to separate files and follow our RBAC best practices.
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

File: stable/metallb/templates/controller.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  namespace: metallb-system
+  name: controller
+  labels:
+    {{- if .Values.helmLabels }}
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
```
Comment (unguiculus) [inline stable/metallb/templates/controller.yaml]: `chart: "{{ template "metallb.chart" . }}"`

File: stable/metallb/templates/controller.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    {{- end }}
+    app: {{ template "metallb.name" . }}
+    component: controller
+  annotations:
```
Comment (unguiculus) [inline stable/metallb/templates/controller.yaml]: Make annotations configurable. You don't need them if you use the Prometheus Operator.

File: stable/metallb/templates/controller.yaml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    prometheus.io/scrape: "true"
+    prometheus.io/port: "7472"
+spec:
+  revisionHistoryLimit: 3
+  selector:
+    matchLabels:
+      app: {{ template "metallb.name" . }}
+      component: controller
+      {{- if .Values.helmLabels }}
```
Comment (unguiculus) [inline stable/metallb/templates/controller.yaml]: Remove `if` statement. Standard labels should always be added.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by danderson** [comment]
Thank you for the review! I made the requested changes, PTAL.

File: stable/metallb/templates/config.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{{- if (or .Values.config .Values.arpCIDR) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: {{ .Release.Namespace | quote }}
```
Comment (unguiculus) [inline stable/metallb/templates/config.yaml]: Just remove the namespace completely. It is unnecessary. The result will be the same. Apply everywhere.

File: stable/metallb/README.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+implementation for LoadBalancer Service objects.
+
+MetalLB is a cluster service, and as such can only be deployed as a
+cluster singleton. Running multiple installations of MetalLB in a
+single cluster is not supported.
+
+Prerequisites
+-------------
+
+-	Kubernetes 1.9+
```
Comment (unguiculus) [inline stable/metallb/README.md:30]: Remove unnecessary spaces.

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Review by unguiculus — APPROVED** [review]

**Pull request merged**